src/inmanta/moduletool.py

```diff
@@ -1741,7 +1741,15 @@ def _move_data_files_into_namespace_package_dir(self, build_path: str) -> None:
         """
         Copy all files that have to be packaged into the Python package of the module
         """
-        python_pkg_dir = os.path.join(build_path, "inmanta_plugins", self._module.name)
+        python_pkg_dir: str = os.path.join(build_path, "inmanta_plugins", self._module.name)
+        model_dir: str = os.path.join(python_pkg_dir, "model")
+        if os.path.exists(model_dir):
+            raise ModuleBuildFailedError(
+                msg="There is already a model directory in %s, which is not supported. "
+                "Please remove the model directory from this location."
```

Comment:
Perhaps a bit more solution-oriented:
```suggestion
                msg="There is already a model directory in %s. The `inmanta_plugins.TODO.model` package is reserved for bundling the inmanta model files."
                "Please use a different name for this Python package."
```

Wdyt? Note the TODO